routes/main.js

```diff
@@ -0,0 +1,218 @@
+const router = require('express').Router();
+const faker = require('faker');
+const Product = require('../models/product');
+const Review = require('../models/review');
+
+router.get('/generate-fake-data', (req, res, next) => {
+	for (let i = 0; i < 90; i++) {
+		let product = new Product();
+
+		product.category = faker.commerce.department();
+		product.name = faker.commerce.productName();
+		product.price = faker.commerce.price();
+		product.image = 'https://via.placeholder.com/250?text=Product+Image';
+
+		product.save();
+	}
+	res.end();
+});
+
+router.get('/products', async (req, res) => {
+	const itemsPerPage = 9;
+	const page = parseInt(req.query.page) || 1;
+	const category = req.query.category;
+	const priceSort = req.query.price;
+	const searchParam = req.query.query;
+
+	let query = {};
+
+	if (category) {
+		query.category = category;
+	}
+
+	if (searchParam) {
+		query.$or = [
+			{ name: { $regex: new RegExp(searchParam, 'i') } }, // Case-insensitive search in names
+			{ category: { $regex: new RegExp(searchParam, 'i') } }, // Case-insensitive search in categories
+		];
+	}
+
+	let sortCriteria = {};
+
+	if (priceSort === 'highest') {
+		sortCriteria = { price: -1 };
+	} else if (priceSort === 'lowest') {
+		sortCriteria = { price: 1 };
+	}
+
+	try {
+		const totalCount = await Product.countDocuments(query);
+		if (totalCount === 0) {
+			return res.status(404).send('No products found');
```

Comment:
is it really and error if no products were found? 
404 is not the right error as it refers to a resource in the url not found, if you make a request for: '/prodcts' i would expect to see 404